akvo/rsr/tests/rest/test_project_update.py

```diff
@@ -200,6 +201,106 @@ def test_rest_post_project_update_photo_none(self):
+        self.update_enum = ProjectUpdate.objects.create(
+            project=self.project,
+            user=self.enumerator,
+            title="My simple title",
+        )
+        self.update_steve = ProjectUpdate.objects.create(
+            project=self.project,
+            user=self.enumerator_steve,
+            title="My second simple title",
+        )
+
+        self.c.force_login(self.enumerator)
+
+    def test_view_all_updates(self):
+        """Ensure all updates can still be read"""
+
+        response = self.c.get("/rest/v1/project_update/?format=json")
+        self.assertEqual(response.status_code, 200, msg=response.content)
+        data = response.json()
+        self.assertEqual(data["count"], 2)
+
+    def test_own_update(self):
+        """Ensure the owned update has certain properties"""
+
+        response = self.c.get(f"/rest/v1/project_update/{self.update_enum.id}/?format=json")
+        self.assertEqual(response.status_code, 200, msg=response.content)
+        data = response.json()
+        self.assertEqual(data["editable"], True)
+        self.assertEqual(data["deletable"], True)
+
+    def test_other_update(self):
+        """Ensure the update owned by the other enumerator has certain properties"""
+
+        response = self.c.get(f"/rest/v1/project_update/{self.update_steve.id}/?format=json")
+        self.assertEqual(response.status_code, 200, msg=response.content)
+        data = response.json()
+
+        self.assertEqual(data["editable"], False)
+        self.assertEqual(data["deletable"], False)
+
+    def test_fail_modify_updates_by_other_enumerators(self):
+        """Modifying updates made by others should fail"""
+
+        response = self.c.patch(
+            f"/rest/v1/project_update/{self.update_steve.id}/?format=json",
+            json.dumps({"title": "new title"}),
+            content_type="application/json"
+        )
+        self.assertEqual(response.status_code, 403, msg=response.content)
+
+    def test_fail_delete_updates_by_other_enumerators(self):
+        """Deleting updates made by others should fail"""
+        response = self.c.delete(
+            f"/rest/v1/project_update/{self.update_steve.id}/?format=json",
+        )
+        self.assertEqual(response.status_code, 403, msg=response.content)
+
+    def test_modify_own_update(self):
+        """Modifying owned updates should succeed"""
+        response = self.c.patch(
+
+            f"/rest/v1/project_update/{self.update_enum.id}/?format=json",
+            json.dumps({"title": "new title"}),
+            content_type="application/json"
+        )
+        self.assertEqual(response.status_code, 200, msg=response.content)
+
+    def test_delete_own_update(self):
+        """Deleting owned updates should succeed"""
+
+        response = self.c.delete(
+            f"/rest/v1/project_update/{self.update_enum.id}/?format=json",
+        )
+        self.assertEqual(response.status_code, 204, msg=response.content)
+
+
```

Comment:
I think it's necessary to add testing for the enumerator role on restricted project access to make sure the permission rules work on both unrestricted (organisation-based) and restricted (project-based) access modes.

See [here](https://github.com/akvo/akvo-rsr/blob/bug/5038-enumerators-can-edit-project-updates-they-do-not-own/akvo/rsr/tests/rest/test_project_enumerators.py#L361-L364) on how to setup a project-based access role.

Reply:
:cry:  There was no clean to solve it that I could think of. The `ProjectRolePermissionBackend` now has a special condition for models that we think should allow a check for ownership. Right now, only `rsr.projectupdate` is considered for the special condition.